app/dts/bindings/zmk,wakeup-trigger-key.yaml

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2023 The ZMK Contributors
+# SPDX-License-Identifier: MIT
+
+description: |
+  Driver for a dedicated key for waking the device from sleep
+
+compatible: "zmk,wakeup-trigger-key"
```

Comment:
If renaming `zmk,behavior-key` to something that includes `gpio-key`, consider renaming this as well.

e.g. if we like `zmk,gpio-key-behavior-trigger`, then this could be `zmk,gpio-key-wakeup-trigger` for consistency.